compiler/resolution/functionResolution.cpp

```diff
@@ -10258,8 +10258,10 @@ static FnSymbol* autoMemoryFunction(AggregateType* at, const char* fnName) {
     }
   }
 
+  /*
   if (retval != NULL)
     INT_ASSERT(retval->hasFlag(FLAG_PROMOTION_WRAPPER) == false);
+  */
```

Comment:
@mppf: What can you tell me about this assertion?  All I know about it so far is that when it was in, I couldn't create hash tables with array elements, and with it out, that seems to work.  But I'd prefer to understand why or what this is guarding against rather than just removing it for that reason.

Reply:
It's trying to guard against initCopy/autoCopy/autoDestroy for an array type resolving to a promotion. In particular we want these to call into the array implementation directly and I would expect incorrect memory management if we use a promotion.

Reply:
So is the idea that if I were to run my branch on one of the tests that triggered this assertion, I'd either see a memory leak or a valgrind/asan failure?  Or something else?

Reply:
I think that's a possibility but even if there is no valgrind/asan failure I would characterize it as a situation where the compiler is getting confused.